internal/cmd/egctl/translate.go

```diff
@@ -644,6 +735,46 @@ func kubernetesYAMLToResources(str string, addMissingResources bool) (*gatewayap
 				resources.Namespaces = append(resources.Namespaces, namespace)
 			}
 		}
+
+		requiredServiceMap := map[string]map[string]*v1.Service{}
```

Comment:
thoughts on using 2 state entities (existingServiceMap, resources.Services)
existingServiceMap := map[string]bool where the key is the namespacedName of the service

So the logic would be - if a backend doesnt exist in map, add it to the map as well as to `resources`

Reply:
Thanks for your suggestion. It indeed simplifies the logic. There is a need to merge multiple ports of the service in the same name:
```
			inserted := false
			for _, port := range service.Spec.Ports {
				if port.Protocol == servicePort.Protocol && port.Port == servicePort.Port {
					inserted = true
					break
				}
			}

			if !inserted {
				service.Spec.Ports = append(service.Spec.Ports, servicePort)
			}
```
So we still need to track the v1.Service.

Reply:
good catch, agree this is needed